### PR TITLE
refactor: Rename identifiers for clarification

### PR DESCRIPTION
**Pull request by shadeMe**
### Proposed Changes:

 
 
Feedback from @bilgeyucel's review of the example notebook. Also includes some test refactoring to avoid unnecessary failures.

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack/blob/main/code_of_conduct.txt)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.
- I documented my code
- I ran [pre-commit hooks](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md#installation) and fixed any issue

### CODE REVIEW

**Comment by shadeMe** [comment]
Test failures are unrelated to the changes in this PR.

File: haystack_experimental/evaluation/harness/rag/evaluation_pipeline.py [hunk 1]
```diff
@@ -44,8 +44,12 @@ def default_rag_evaluation_pipeline(
         RAGEvaluationMetric.SEMANTIC_ANSWER_SIMILARITY: partial(
             SASEvaluator, model="sentence-transformers/all-MiniLM-L6-v2"
         ),
-        RAGEvaluationMetric.ANSWER_FAITHFULNESS: partial(FaithfulnessEvaluator, raise_on_failure=False),
-        RAGEvaluationMetric.CONTEXT_RELEVANCE: partial(ContextRelevanceEvaluator, raise_on_failure=False),
+        RAGEvaluationMetric.FAITHFULNESS: partial(
```
Comment (davidsbatista) [inline haystack_experimental/evaluation/harness/rag/evaluation_pipeline.py:47]: with ruff line length = 120 this should fit on on-line

**Review by davidsbatista — APPROVED** [review]
mostly name changes, LGTM

File: haystack_experimental/evaluation/harness/rag/evaluation_pipeline.py (hunk 1 above)
Comment (shadeMe) [reply haystack_experimental/evaluation/harness/rag/evaluation_pipeline.py:47]: I've leave that for another PR.

**Pull request merged**